src/psyclone/psyir/backend/sir.py

```diff
@@ -345,7 +359,15 @@ def reference_node(self, node):
             raise VisitorError(
                 "Method reference_node in class SIRWriter: SIR Reference "
                 "node is not expected to have any children.")
-        return "{0}make_var_access_expr(\"{1}\")".format(self._nindent,
+        # _scalar_names is a set so duplicates will be ignored. It
```

Comment:
So does `make_var_access_expr` not existing in the SIR API?

Reply:
Yes it does exist, we just don't make use of it at the moment. The reason for this is that it is unclear how to declare scalars in the SIR and the proposed solution (from the Dawn developers) was to specify these as temporary fields (which is easy) as the Dawn backend translates temporary fields to the lowest possible dimension i.e. a scalar in our case.

Reply:
OK. Thanks for explaining.